### PR TITLE
add whitelist_for_serdes to backfill related classes

### PR DESCRIPTION
**Pull request by ruizh22**
## Summary & Motivation
Need to whitelist `BackfillPolicy` and classes needed for ranged partition backfill, otherwise, we will get errors like
`dagster._serdes.errors.SerializationError: Can only serialize whitelisted namedtuples, received BackfillPolicy(max_partitions_per_run=None).`
when trying to deploy the user code.

## How I Tested These Changes
No unit test covering this whitelisting feature.

### CODE REVIEW

**Comment by jamiedemaria** [comment]
@alangenfeld mind taking a lok?

**Comment by alangenfeld** [comment]
can you update one of the backfill tests in https://github.com/dagster-io/dagster/blame/master/python_modules/dagster/dagster_tests/core_tests/host_representation_tests/test_external_data.py to round trip one of the `ExternalAssetNode` s with `BackfillPolicy` through serialization? Use the `serialize_value` / `deserialize_value` from `dagster._serdes.serdes`

File: python_modules/dagster/dagster_tests/core_tests/host_representation_tests/test_external_data.py [hunk 1]
```diff
@@ -118,6 +118,8 @@ def asset1():
         )
     ]
 
+    deserialize_value(serialize_value(external_asset_nodes[0]), ExternalAssetNode)
```
Comment (alangenfeld) [inline python_modules/dagster/dagster_tests/core_tests/host_representation_tests/test_external_data.py:121]: assert that the values before and after are `==`

**Review by alangenfeld — APPROVED** [review]
one small test update before landing but otherwise good to go

**Pull request merged**